src/OpenApi/gen/Helpers/ISymbolExtensions.cs

```diff
@@ -189,4 +189,26 @@ public static IEnumerable<INamedTypeSymbol> GetBaseTypes(this ITypeSymbol? type)
             current = current.BaseType;
         }
     }
+
+    public static bool IsAccessibleType(this ISymbol symbol)
```

Comment:
This helps us avoid generating XML comments for non-public types and internal properties in non-public types.

Note: this does mean that DTOs have to be defined as public types if they are defined in the same assembly as the API (see where this is called in the `AssemnlyTypeSymbolVisitor` above). We can consider changing this behavior to allow XML comments on internal types in the application assembly.

Reply:
Seems a bit overaggressive. I'm guessing we'll change this behavior a bit in the future.

Any reason to disallow the type entirely just because it has internal types?

Reply:
I just hit this error in preview 2:

```text
C:\Coding\martincostello\alexa-london-travel-site\artifacts\obj\LondonTravel.Site\debug\Microsoft.AspNetCore.OpenApi.SourceGenerators\Microsoft.AspNetCore.OpenApi.SourceGenerators.XmlCommentGenerator\OpenApiXmlCommentSupport.generated.cs(207,125): error CS0122: 'CustomHttpHeadersMiddleware.Csp' is inaccessible due to its protection level
```

This relates to a `private` nested class in `public` class that has some `///` comments on the `internal const string` properties in it.

Would this change fix that scenario too?

Reply:
One more scenario for you - seems like it's trying to document things to do with the Regex source generator ([example](https://github.com/martincostello/adventofcode/blob/9d9501cbd871409db7bb6e7f38a2dbccd10ef76e/src/AdventOfCode/Puzzles/Y2020/Day04.cs#L29-L30)):

```text
C:\Coding\martincostello\adventofcode\artifacts\obj\AdventOfCode.Site\debug\Microsoft.AspNetCore.OpenApi.SourceGenerators\Microsoft.AspNetCore.OpenApi.SourceGenerators.XmlCommentGenerator\OpenApiXmlCommentSupport.generated.cs(1589,94): error CS0234: The type or namespace name 'HexColor_1' does not exist in the namespace 'System.Text.RegularExpressions.Generated' (are you missing an assembly reference?)
```

If not catered for by this change, I can open a new issue.

Reply:
It depends on whether the type in question is defined in or outside the assembly.

I believe these bits have landed in the nightly preview3 SDK if you want to take another bump to try it out.

Alternatively, I'd recommend filing an issue so we can at least capture this scenario in a test case even if it is already fixed.

Reply:
I'll try out a nightly tomorrow and see if you think it should be fixed.

Reply:
Confirmed these two are catered for as of `10.0.100-preview.3.25169.19`, but confirmed #61019 is still an issue and found two new ones 😅. Issues shortly...

Reply:
I think they're probably the same one issue: #61035

Reply:
> I believe these bits have landed in the nightly preview3 SDK if you want to take another bump to try it out.

Alternatively, I'd recommend filing an issue so we can at least capture this scenario in a test case even if it is already fixed.

Found a different case where a private type causes a compiler error.

Reply:
#61037